plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java

```diff
@@ -117,21 +121,25 @@ private void process(Plugin plugin) {
             // Minimum JDK to run openrewrite
             plugin.withJDK(JDK.JAVA_17);
 
-            // Collect metadata
-            plugin.collectMetadata(mavenInvoker);
-
-            // Move metadata from the target directory of the plugin to the common cache
-            CacheManager pluginCacheManager = new CacheManager(Path.of(Settings.TEST_PLUGINS_DIRECTORY)
-                    .resolve(plugin.getLocalRepository().resolve("target")));
-            plugin.setMetadata(pluginCacheManager.move(
-                    cacheManager,
-                    Path.of(plugin.getName()),
-                    CacheManager.PLUGIN_METADATA_CACHE_KEY,
-                    new PluginMetadata(pluginCacheManager)));
-            LOG.debug(
-                    "Moved plugin {} metadata to cache: {}",
-                    plugin.getName(),
-                    plugin.getMetadata().getLocation().toAbsolutePath());
+            // Collect metadata and move metadata from the target directory of the plugin to the common cache
+            if (plugin.getMetadata() == null) {
+
+                plugin.collectMetadata(mavenInvoker);
+
+                CacheManager pluginCacheManager = new CacheManager(Path.of(Settings.TEST_PLUGINS_DIRECTORY)
```

Comment:
I'm not happy with this code. This probably need to be moved somewere (even a private method would fix). Will clean on an other PR